.github/workflows/test.yml

```diff
@@ -0,0 +1,111 @@
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Go test
+    runs-on: ubuntu-latest
+    if: ${{ github.event.pull_request.head.repo.full_name == github.repository }}
+    environment: prod
+    permissions:
+      id-token: write
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get vault secrets
+        id: vault_secrets
+        uses: hashicorp/vault-action@7709c609789c5e27b757a85817483caadbb5939a # v3.3.0
+        with:
+          method: jwt
+          url: ${{ vars.VAULT_ADDR }}
+          path: ${{ vars.VAULT_AUTH_PATH }}
+          role: ${{ vars.VAULT_ROLE }}
+          secrets: |
+            SHARED/data/common SONAR_ORG_GH | SONAR_ORG ;
+            SHARED/data/common SONAR_LOGIN_GH | SONAR_TOKEN ;
+            SHARED/data/common SONAR_KEY_GH | SONAR_KEY ;
+            SHARED/data/common SONAR_HOST | SONAR_HOST ;
+            SHARED/data/common DATADOG_API_KEY | DATADOG_API_KEY ;
+
+      - name: Setup Go
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version-file: "go.mod"
+
+      - name: Install linters
+        run: go install golang.org/x/lint/golint@latest
+
+      - name: Go format and lint
+        run: |
+          go fmt ./...
+          golint ./...
+          go vet ./...
+
+      - name: Suggest changes
+        uses: reviewdog/action-suggester@4747dbc9f9e37adba0943e681cc20db466642158 # v1.21.0
+        with:
+          tool_name: go static analysis
+
+      - name: Test
+        uses: robherley/go-test-action@b19f6aadabfb1ad85079065b21aa2af132466468 # v0.6.0
+
+      - name: Cache SonarQube packages
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 #v4.2.3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: SonarQube scan
+        uses: SonarSource/sonarqube-scan-action@2500896589ef8f7247069a56136f8dc177c27ccf # v5.2.0
+        with:
+          args: >
+            -Dsonar.host.url=${{ steps.vault_secrets.outputs.SONAR_HOST }}
+            -Dsonar.token=${{ steps.vault_secrets.outputs.SONAR_TOKEN }}
+            -Dsonar.organization=${{ steps.vault_secrets.outputs.SONAR_ORG }}
+            -Dsonar.projectKey=${{ steps.vault_secrets.outputs.SONAR_KEY }}_${{ github.event.repository.name }}
+            -Dsonar.projectName=${{ github.event.repository.name }}
+            -Dsonar.links.scm="https://github.com/${{ github.repository }}"
+            -Dsonar.scm.provider=git
+            ${{ format('-Dsonar.pullrequest.branch={0} -Dsonar.pullrequest.key={1}', github.head_ref || github.ref_name, github.event.pull_request.number) }}
+
+      - name: Submit Datadog metrics
```

Comment:
These metrics is meant for us to be able to track the usage of our reusable workflow. So let's remove this step from here